test/test_git.py

```diff
@@ -102,9 +90,26 @@ def test_it_executes_git_not_from_cwd(self):
                     print("#!/bin/sh", file=file)
                 os.chmod(impostor_path, 0o755)
 
-            with _chdir(tmpdir):
+            with cwd(tmpdir):
                 self.assertRegex(self.git.execute(["git", "version"]), r"^git version\b")
 
+    @skipUnless(is_win, "The regression only affected Windows, and this test logic is OS-specific.")
+    def test_it_avoids_upcasing_unrelated_environment_variable_names(self):
+        old_name = "28f425ca_d5d8_4257_b013_8d63166c8158"
```

Comment:
I see, this variable name is used to assure nobody will possibly stumble over it, nor is it likely to clash with an existing variable.
I think it's fair as nobody is iterating environment variables either.

Reply:
Another reason I named it like that is that it sticks around after the test, because I don't think there's a reasonable way to unset it (though there are probably some unreasonable ways). Setting the variable with `os.putenv` sets it case-sensitively on all systems including Windows, but it is not reflected in `os.environ` (and `nt.environ` only has the variables as they existed when the process started).